tutorials/examples/train_conditional.py

```diff
@@ -0,0 +1,236 @@
+        output_dim=1,
+        hidden_dim=16,
+        n_hidden_layers=2,
+    )
+
+    logZ_estimator = ScalarEstimator(module_logZ)
+    gflownet = TBGFlowNet(logZ=logZ_estimator, pf=pf_estimator, pb=pb_estimator)
+
+    return gflownet
+
+
+def build_db_gflownet(env):
+    pf_estimator, pb_estimator = build_conditional_pf_pb(env)
+    logF_estimator = build_conditional_logF_scalar_estimator(env)
+    gflownet = DBGFlowNet(logF=logF_estimator, pf=pf_estimator, pb=pb_estimator)
+
+    return gflownet
+
+
+def build_db_mod_gflownet(env):
+    pf_estimator, pb_estimator = build_conditional_pf_pb(env)
+    gflownet = ModifiedDBGFlowNet(pf=pf_estimator, pb=pb_estimator)
+
+    return gflownet
+
+
+def build_fm_gflownet(env):
+    CONCAT_SIZE = 16
+    module_logF = NeuralNet(
+        input_dim=env.preprocessor.output_dim,
+        output_dim=CONCAT_SIZE,
+        hidden_dim=256,
+    )
+    module_cond = NeuralNet(
+        input_dim=1,
+        output_dim=CONCAT_SIZE,
+        hidden_dim=256,
+    )
+    module_final_logF = NeuralNet(
+        input_dim=CONCAT_SIZE * 2,
+        output_dim=env.n_actions,
+    )
+    logF_estimator = ConditionalDiscretePolicyEstimator(
+        module_logF,
+        module_cond,
+        module_final_logF,
+        env.n_actions,
+        is_backward=False,
+        preprocessor=env.preprocessor,
+    )
+
+    gflownet = FMGFlowNet(logF=logF_estimator)
+
+    return gflownet
+
+
+def build_subTB_gflownet(env):
+    pf_estimator, pb_estimator = build_conditional_pf_pb(env)
+    logF_estimator = build_conditional_logF_scalar_estimator(env)
+    gflownet = SubTBGFlowNet(logF=logF_estimator, pf=pf_estimator, pb=pb_estimator)
+
+    return gflownet
+
+
+def train(env, gflownet):
+
+    torch.manual_seed(0)
+    exploration_rate = 0.5
+    lr = 0.0005
+
+    # Move the gflownet to the GPU.
+    if torch.cuda.is_available():
+        gflownet = gflownet.to("cuda")
+
+    # Policy parameters and logZ/logF get independent LRs (logF/Z typically higher).
+    if type(gflownet) is TBGFlowNet:
+        optimizer = torch.optim.Adam(gflownet.pf_pb_parameters(), lr=lr)
+        optimizer.add_param_group({"params": gflownet.logz_parameters(), "lr": lr * 100})
+    elif type(gflownet) is DBGFlowNet or type(gflownet) is SubTBGFlowNet:
+        optimizer = torch.optim.Adam(gflownet.pf_pb_parameters(), lr=lr)
+        optimizer.add_param_group({"params": gflownet.logF_parameters(), "lr": lr * 100})
+    elif type(gflownet) is FMGFlowNet or type(gflownet) is ModifiedDBGFlowNet:
+        optimizer = torch.optim.Adam(gflownet.parameters(), lr=lr)
+    else:
+        print("What is this gflownet? {}".format(type(gflownet)))
+
+    n_iterations = int(10)  #1e4)
+    batch_size = int(1e4)
+
+    print("+ Training Conditional {}!".format(type(gflownet)))
+    for i in (pbar := tqdm(range(n_iterations))):
+        conditioning = torch.rand((batch_size, 1))
+        conditioning = (conditioning > 0.5).to(torch.float)  # Randomly 1 and zero.
```

Comment:
IIUC, the conditioning doesn't change anything in this example. While this file is a great way to show how one can code their conditional gflownet, what do you think of slightly altering the setting here, e.g. by making the environment conditional (e.g. hide one of the 4 modes if conditioning=1), and then, post-training, have some validation where we compare the resulting pair of distributions to the pair of target distributions ?

Reply:
yes, you're right. can we save this for a follow up PR?

Reply:
I filed the issue here. if you agree I'd like to do this work separately. 

https://github.com/GFNOrg/torchgfn/issues/190